gcsio/src/main/java/com/google/cloud/hadoop/gcsio/FolderInfo.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2013 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.gcsio;
+
+import com.google.api.services.storage.model.Folder;
+
+/** Contains information about a Folder resource */
+public class FolderInfo {
```

Comment:
Can this be not public?

Reply:
This has to be  public for it's utilization

Reply:
Is this made public to enable testing? If testing is the only reason, annotate this with VisibleForTesting

Reply:
No testing is not the only reason. The implementation is such that after listing, all objects are converted into FolderInfo class and even for deletion we require a list of FolderInfo objects. 

Still annotated for testing. Hope I was clear.

Reply:
It's there in testing package also.